### PR TITLE
Tried to created delegate and event.

### PR DESCRIPTION
**Pull request by MindaugasJusis**
Tried to created delegate and event.

### CODE REVIEW

File: FoodWaste/Controllers/DataBaseOperations.cs [hunk 1]
```diff
@@ -26,6 +27,25 @@ public static async Task<List<Product>> GetProduct()
         public static async Task<List<Product>> UpdateProducts()
         {
             List<Product> products = new List<Product>();
```
Comment (PauliusKibilda) [inline FoodWaste/Controllers/DataBaseOperations.cs:29]: this shouldnt exist

File: FoodWaste/Controllers/DataBaseOperations.cs [hunk 2]
```diff
@@ -26,6 +27,25 @@ public static async Task<List<Product>> GetProduct()
...
+            {
+                if (e.Action == System.Collections.Specialized.NotifyCollectionChangedAction.Reset)
+                {
+
+                    foreach (Product prod in myproducts)
+                    {
+                        if (prod.ExpiryDate < DateTime.Today)
+                        {
+                            prod.State.Equals("Expired");
+                            PutProduct(prod);
```
Comment (VaidasCib) [inline FoodWaste/Controllers/DataBaseOperations.cs:43]: make delegate async and await PutProduct(prod);

File: FoodWaste/Controllers/DataBaseOperations.cs [hunk 3]
```diff
@@ -26,6 +27,25 @@ public static async Task<List<Product>> GetProduct()
...
+            delegate (object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+            {
+                if (e.Action == System.Collections.Specialized.NotifyCollectionChangedAction.Reset)
+                {
+
+                    foreach (Product prod in myproducts)
+                    {
+                        if (prod.ExpiryDate < DateTime.Today)
+                        {
+                            prod.State.Equals("Expired");
```
Comment (VaidasCib) [inline FoodWaste/Controllers/DataBaseOperations.cs:42]: this line does nothing, change to prod.State = Product.ProductState.Expired;

File: FoodWaste/Controllers/DataBaseOperations.cs [hunk 4]
```diff
@@ -26,6 +27,25 @@ public static async Task<List<Product>> GetProduct()
...
         {
             List<Product> products = new List<Product>();
+
+            ObservableCollection<Product> myproducts = new ObservableCollection<Product>();
+            myproducts.CollectionChanged += new System.Collections.Specialized.NotifyCollectionChangedEventHandler(
+            delegate (object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+            {
+                if (e.Action == System.Collections.Specialized.NotifyCollectionChangedAction.Reset)
+                {
+
```
Comment (VaidasCib) [inline FoodWaste/Controllers/DataBaseOperations.cs:37]: inconsistent spacing

**Comment by VaidasCib** [comment]
> "make delegate async and await PutProduct(prod)"

I thought I needed to do this at first, but it caused some unexpected exceptions, so I just left it to run synchronously

**Pull request merged**